libgalois/include/katana/GraphTopology.h

```diff
@@ -783,12 +790,18 @@ class KATANA_EXPORT ProjectedTopology : public GraphTopologyTypes {
   NUMAArray<Node> projected_to_original_nodes_mapping_;
   NUMAArray<Edge> original_to_projected_edges_mapping_;
   NUMAArray<Edge> projected_to_original_edges_mapping_;
+  NUMAArray<uint8_t> node_bitmask_;
+  NUMAArray<uint8_t> edge_bitmask_;
 };
 
 template <typename Topo>
 class KATANA_EXPORT ProjectedTopologyWrapper : public GraphTopologyTypes {
+  using Base = Topo;
+
 public:
-  explicit ProjectedTopologyWrapper(const Topo* t) noexcept : topo_ptr_(t) {
+  explicit ProjectedTopologyWrapper(
+      const PropertyGraph* pg, const Topo* t) noexcept
```

Comment:
Will having a topology that depends on a graph create any weird cyclical references or anything? It may not. I just want to check.

Reply:
Agree. Please move this class out of this file and change the class name to `ProjectedPropertyGraphWrapper`.  

Why are `PropertyGraph` and `Topo` passed and stored? Doesn't `PropertyGraph` have the `Topo`? This also leads to the dependent arguments problem I mentioned. What if the topology in `PropertyGraph` is NOT the same as the `Topo` passed?

Reply:
so which class should I put this into? `PropertyGraph.h` one? I did the same thign as Amber had for SortedTopology thing, after consulting him.

Reply:
If it has properties, it is no longer a simple `*Topology`. 

@amberhassaan Where should this go? I'm fine if it's in a new header.